Seasonal/Mogloween/MergeShops/TricktownMerge.cs

```diff
@@ -0,0 +1,126 @@
+//cs_include Scripts/Seasonal/Mogloween/CoreMogloween.cs
+using Skua.Core.Interfaces;
+using Skua.Core.Models.Items;
+using Skua.Core.Options;
+
+public class TricktownMerge
+{
+    public IScriptInterface Bot => IScriptInterface.Instance;
+    public CoreBots Core => CoreBots.Instance;
+    public CoreFarms Farm = new();
+    public CoreStory Story = new();
+    public CoreAdvanced Adv = new();
+    public static CoreAdvanced sAdv = new();
+    public CoreMogloween CoreMogloween = new();
+
+    public List<IOption> Generic = sAdv.MergeOptions;
+    public string[] MultiOptions = { "Generic", "Select" };
+    public string OptionsStorage = sAdv.OptionsStorage;
+    // [Can Change] This should only be changed by the author.
+    //              If true, it will not stop the script if the default case triggers and the user chose to only get mats
+    private bool dontStopMissingIng = false;
+
+    public void ScriptMain(IScriptInterface bot)
+    {
+        Core.BankingBlackList.AddRange(new[] { "Treats", "Ghastly Gummy "});
+        Core.SetOptions();
+
+        BuyAllMerge();
+
+        Core.SetOptions(false);
+    }
+
+    public void BuyAllMerge()
+    {
+        if (!Core.isSeasonalMapActive("tricktown"))
+            return;
+        CoreMogloween.TrickTown();
+        
+        //Only edit the map and shopID here
+        Adv.StartBuyAllMerge("tricktown", 2179, findIngredients);
+
+        #region Dont edit this part
+        void findIngredients()
+        {
+            ItemBase req = Adv.externalItem;
+            int quant = Adv.externalQuant;
+            int currentQuant = req.Temp ? Bot.TempInv.GetQuantity(req.Name) : Bot.Inventory.GetQuantity(req.Name);
+            if (req == null)
+            {
+                Core.Logger("req is NULL");
+                return;
+            }
+
+            switch (req.Name)
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
+                    break;
+                #endregion
+
+                case "Treats":
+                    Core.FarmingLogger(req.Name, quant);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {
+                        Core.AddDrop("Treats");
+                        Core.Join("tricktown");
+                        Core.KillMonster("trickortreat", "Enter", "Spawn", "Trick or Treater");
+                        Bot.Wait.ForPickup(req.Name);
+                    }
+                    break;
+
+                case "Ghastly Gummy":
+                    Core.FarmingLogger(req.Name, quant);
+                    Core.RegisterQuests(8936);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {

```

Comment:
Add: 
Core.EquipClass(ClassType.Solo); for the hard monster 
and 
Core.EquipClass(ClassType.Farm);
for the regualr ones